database/migration/migrateData.ts

```diff
@@ -496,11 +496,48 @@ const migrateData = async () => {
       SET NOT NULL;
     `)
 
+    console.log(' - Add "user_id" field to trigger_schedule to track user who made changes')
+
+    await DB.changeSchema(`
+      ALTER TABLE trigger_schedule
+      ADD COLUMN editor_user_id integer REFERENCES public.user (id) ON DELETE CASCADE;
+    `)
+
     console.log(' - Adding applicant_deadline to application_list')
 
+    // We need to drop and re-create the application_list_shape table so the
+    // column order gets preserved, otherwise the subsequent application_list
+    // function will fail
     await DB.changeSchema(`
-      ALTER TABLE application_list_shape
-      ADD COLUMN IF NOT EXISTS applicant_deadline timestamptz;
+      DROP TABLE IF EXISTS application_list_shape CASCADE;
```

Comment:
probably should be in the next statement. Also would this be allowed, since application list uses it ? Should we drop application list first ?

Reply:
The `CASCADE` forces the application_list() function to also be dropped. And then they both get recreated in the next bit.